pretyping/pretyping.ml

```diff
@@ -378,10 +378,10 @@ let check_evars_are_solved env current_sigma init_sigma =
   let frozen = frozen_and_pending_holes (init_sigma, current_sigma) in
   check_evars_are_solved env current_sigma frozen
 
-let process_inference_flags flags env initial_sigma (sigma,c) =
+let process_inference_flags flags env initial_sigma (sigma,c,cty) =
   let sigma = solve_remaining_evars flags env sigma initial_sigma in
   let c = if flags.expand_evars then nf_evar sigma c else c in
```

Comment:
Should we not nf_evar cty?

Reply:
Good spot.  I did it on purpose, since the API is new nobody can rely on the fact that the type is substituted. I did not write (in a comment) it because it is an EConstr, so you can't access it without using the EConstr API that applies the substitution on the fly. IMHO the fact that we return an EConstr (for the term) and that we have a flag to apply the substitution is a very fishy thing. @ppedrot  what do you think?

Reply:
Removing the normalization has a non-trivial consequence in term of performance of a few developments AFAIR.